common/src/main/java/org/apache/uniffle/common/ReceivingFailureServer.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.uniffle.common.rpc.StatusCode;
+import org.apache.uniffle.proto.RssProtos;
+
+public class ReceivingFailureServer {
```

Comment:
Could we give this class a better name? If this class will be reused for rebalance, this may not a good name. Because the high load may just trigger a redirect instead of error.

Reply:
The server can receive failure or rejection. Maybe we should find the common case for failure or rejection.

Reply:
Maybe rejection can be looked a kind of failure. If we use rejection instead of redirect. Maybe we can use this name.

Reply:
let's reserve this name until having another better name.